### PR TITLE
Update user style

### LINKED ISSUE #27: BE - Add a mutation and resolver to receive the survey results and update the style of the user 8/26

### PR DESCRIPTION
**Pull request by jpc20**
**What's this PR do?**
We set up the endpoint to update the user's style after they fill out the survey.

**Where should the reviewer start?**
`api/modules/user/mutations.test.js` `api/modules/user/mutations.js`  `api/modules/user/resolvers.js`

**How should this be manually tested?**
This can be tested by running the API, going to `localhost:8000`, and manually entering the mutation.

closes #27

### CODE REVIEW

File: code/api/src/modules/user/mutations.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      const token = resp.body.data.userLogin.token;
+
+      const response = await request(server)
+        .post('/')
+        .set('Authorization', `Bearer ${token}`)
+        .send({ query: `mutation { addStyleToUser(surveyResults: ${JSON.stringify(survey)}) { id name email style } }`})
+        .expect(200)
+
+        expect(response.body.data.addStyleToUser.style).toEqual('punk')
+  })
```
Comment (Lithnotep) [inline code/api/src/modules/user/mutations.test.js:60]: Great job for our first time in jest

File: code/api/src/modules/user/resolvers.js [hunk 2]
```diff
@@ -77,3 +77,14 @@ export async function remove(parentValue, { id }) {
...
+export async function updateStyle(parentValue, { surveyResults }, { auth }) {
+  if(auth.user && auth.user.id > 0) {
+    const user = await models.User.findOne({ where: { id: auth.user.id } })
+    const results = JSON.parse(surveyResults)[0];
+    const newStyle = Object.keys(results).sort((a, b) => results[b] - results[a])[0];
+    return await user.update({style: newStyle})
+  } else {
+    throw new Error('Please login to update your style.')
+  }
+}
```
Comment (Lithnotep) [inline code/api/src/modules/user/resolvers.js:90]: nice work figuring out auth.user

**Review by Lithnotep — APPROVED** [review]

File: code/api/src/modules/user/mutations.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      '/',
+      graphqlHTTP(request => ({
+        schema: schema,
+        graphiql: false,
+        context: {
+          auth: {
+            user: request.user,
+            isAuthenticated: request.user && request.user.id > 0
+          }
+        }
```
Comment (AntonioJacksonII) [inline code/api/src/modules/user/mutations.test.js:25]: Nice job getting this user authentication set up for testing!

File: code/api/src/modules/user/mutations.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      const id = response.body.data.userSignup.id;
+
+      await request(server)
+        .post('/')
+        .send({ query: `mutation { userRemove(id: ${id}) { id name email } }`})
+        .expect(200)
+  })
+
+  it('addStyleToUser', async () => {
+      const survey = '[{"classy":5,"artsy":2,"punk":7,"sporty":6,"nature":3}]';
```
Comment (jpc20) [inline code/api/src/modules/user/mutations.test.js:45]: This is the only format we could get working... hopefully we can get it to work with the actual survey results

File: code/api/src/modules/user/mutations.js [hunk 5]
```diff
@@ -38,3 +38,15 @@ export const userRemove = {
...
+// add style
+export const addStyleToUser = {
+  type: UserType,
+  args: {
+    surveyResults: {
+      name: 'surveyResults',
+      type: GraphQLString
+    }
+  },
+  resolve: updateStyle
```
Comment (jpc20) [inline code/api/src/modules/user/mutations.js:51]: The resolver is where the actual updating of the user takes place.

**Review by AntonioJacksonII — APPROVED** [review]

File: code/api/src/modules/user/resolvers.js [hunk 6]
```diff
@@ -77,3 +77,14 @@ export async function remove(parentValue, { id }) {
 export async function getGenders() {
   return Object.values(params.user.gender)
 }
+
+export async function updateStyle(parentValue, { surveyResults }, { auth }) {
+  if(auth.user && auth.user.id > 0) {
+    const user = await models.User.findOne({ where: { id: auth.user.id } })
+    const results = JSON.parse(surveyResults)[0];
+    const newStyle = Object.keys(results).sort((a, b) => results[b] - results[a])[0];
+    return await user.update({style: newStyle})
```
Comment (garnetred) [inline code/api/src/modules/user/resolvers.js:86]: This looks great! Melissa and I will use this to retrieve the style object from the API.

**Review by garnetred — APPROVED** [review]

File: code/api/src/modules/user/resolvers.js (hunk 2 above)
Comment (Melizzo) [reply code/api/src/modules/user/resolvers.js:90]: Error handling FTW!

File: code/api/src/modules/user/mutations.test.js (hunk 4 above)
Comment (Melizzo) [reply code/api/src/modules/user/mutations.test.js:45]: We can ensure that it's in the right format to send to you.

**Pull request merged**